File: problemas/estruturas-e-bibliotecas/1244/1244.py
def comp(a, b):
    return len(b) - len(a)


def insertionSort(V):
    for i in range(1, len(V)):
        j = i
        k = j - 1

        while(k > -1 and comp(V[j], V[k]) < 0):
            V[j], V[k] = V[k], V[j]
            j -= 1
            k -= 1


N = int(input())

for _ in range(N):
    palavras = input().strip().split(' ')

    insertionSort(palavras)

    print(' '.join(palavras))

File: problemas/estruturas-e-bibliotecas/1244/1244.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

const comp = (a, b) => b.length - a.length;

const insertionSort = (V) => {
    for (let i = 1; i < V.length; ++i){
        let j = i, k = j - 1;

        while(k > -1 && comp(V[j], V[k]) < 0){
            [V[k], V[j]] = [V[j], V[k]];
            --j, --k;
        }
    }
};

let N = parseInt(lines.shift().trim());

for(let i = 0; i < N; ++i){
    let palavras = lines.shift().trim().split(' ');

    insertionSort(palavras);

    console.log(palavras.join(' '));
}
